ruby/ext/google/protobuf_c/map.c

```diff
@@ -388,6 +388,10 @@ VALUE Map_index(VALUE _self, VALUE key) {
 VALUE Map_index_set(VALUE _self, VALUE key, VALUE value) {
   Map* self = ruby_to_Map(_self);
 
+  if (rb_obj_frozen_p(_self)) {
```

Comment:
I believe this should raise a `FrozenError`, not an `ArgumentError`. You might be able to replace this whole conditional with the following:

```c
rb_check_frozen(_self);
```